spec/rspec/core/formatters/snippet_extractor_spec.rb

```diff
@@ -250,6 +250,24 @@ def another_expression(*)
         end
       end
 
+      context 'when Ripper cannot parse the source', :isolated_directory do
+        let(:file_path) do
+          'invalid_source.rb'
+        end
+
+        let(:line_number) do
+          1
+        end
+
+        before do
+          File.write(file_path, 'invalid_source]')
```

Comment:
You've only written one line to the file, which means the test could pass if the implementation instead returned all of the file contents or some constant number of lines.  I think it would make the test more robust if you wrote multiple lines to the file and than showed that it only returned one.  Maybe something like this?

``` ruby
expect("some string").to include(
  "some", "string"
))
```

...and then the expectation can be that it returns `expect("some string").to include(`.